### PR TITLE
Reference best practices in the docs, add "not_in" and "not_has" methods, add "case_sensitive" parameter to "is_in" method

### PR DESCRIPTION
**Pull request by ymost**
Part of #90

### CODE REVIEW

**Review by amos-rimon — APPROVED** [review]

File: pykusto/_src/expressions.py [hunk 1]
```diff
@@ -97,18 +97,39 @@ def __eq__(self, other: _ExpressionType) -> '_BooleanExpression':
...
+        Best practices (`full list <https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/best-practices>`_):
+
+        * Use `case_sensitive=True` when possible
+        """
+        if isinstance(other, (List, Tuple)):
+            # For a literal array, we can use 'in'
+            # The following RHS is the only place where a literal list does not require being surrounded by 'dynamic()'
+            return _BooleanExpression(KQL(f'{self.kql} {"in" if case_sensitive else "in~"} ({", ".join(map(_to_kql, other))})'))
+        # Otherwise, for some reason Kusto does not accept 'in', and we need to use 'contains' as if 'other' was a string
+        return _BooleanExpression.binary_op(other, ' contains_cs ' if case_sensitive else ' contains ', self)
```
[the remark this answers is not in the document]
Comment (ymost) [reply pykusto/_src/expressions.py:115]: I used 'contains' to workaround what seems to be very weird behavior by Kusto, but upon further investigation I still don't know how to work around it.
See here: https://github.com/MicrosoftDocs/dataexplorer-docs/issues/494

File: pykusto/_src/expressions.py [hunk 2]
```diff
@@ -415,12 +450,22 @@ def matches(self, regex: _StringType) -> '_BooleanExpression':
     def contains(self, other: _StringType, case_sensitive: bool = False) -> _BooleanExpression:
         """
         https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/datatypes-string-operators
+
+        Best practices (`full list <https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/best-practices>`_):
+
+        * When looking for full tokens, `has` works better, since it doesn't look for substrings.
```
[the remark this answers is not in the document]
Comment (ymost) [reply pykusto/_src/expressions.py:456]: Too intrusive IMO

File: test/test_expressions.py [hunk 3]
```diff
@@ -387,21 +387,37 @@ def test_column_with_dot(self):
...
+
     def test_is_in_expression(self):
         self.assertEqual(
-            ' | where arrayField contains stringField',
-            Query().where(t.stringField.is_in(t.arrayField)).render()
+            ' | where arrayField contains_cs stringField',
+            Query().where(t.stringField.is_in(t.arrayField, True)).render()
+        )
+
+    def test_not_in_expression(self):
```
[the remark this answers is not in the document]
Comment (ymost) [reply test/test_expressions.py:417]: Added

File: pykusto/_src/expressions.py [hunk 4]
```diff
@@ -462,6 +513,14 @@ def has(self, exp: _StringType, case_sensitive: bool = False) -> '_BooleanExpres
             f'{self.as_subexpression()} {"has_cs" if case_sensitive else "has"} {_to_kql(exp, True)}'
         ))
 
+    def not_has(self, exp: _StringType, case_sensitive: bool = False) -> '_BooleanExpression':
```
[the remark this answers is not in the document]
Comment (ymost) [reply pykusto/_src/expressions.py]: Renamed

File: test/test_expressions.py [hunk 5]
```diff
@@ -410,6 +426,12 @@ def test_has(self):
             Query().where(t.stringField.has("test")).render()
         )
 
+    def test_not_has(self):
```
[the remark this answers is not in the document]
Comment (ymost) [reply test/test_expressions.py]: Added

File: pykusto/_src/expressions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ymost) [reply pykusto/_src/expressions.py]: It turns out we need to use "set_has_element". Fixed

File: pykusto/_src/expressions.py [hunk 6]
```diff
@@ -439,12 +486,18 @@ def endswith(self, other: _StringType, case_sensitive: bool = False) -> _Boolean
     def lower(self) -> '_StringExpression':
         """
         https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/tolowerfunction
+
+        Don't user this method for case-insensitive comparisons. Instead use the `equals` method with `case_sensitive=False`
```
Comment (amos-rimon) [inline pykusto/_src/expressions.py:490]: typo - use instead of user

File: pykusto/_src/expressions.py [hunk 7]
```diff
@@ -439,12 +486,18 @@ def endswith(self, other: _StringType, case_sensitive: bool = False) -> _Boolean
...
+        (see `best practices <https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/best-practices>`_).
         """
         return _StringExpression(KQL(f'tolower({self.kql})'))
 
     def upper(self) -> '_StringExpression':
         """
         https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/toupperfunction
+
+        Don't user this method for case-insensitive comparisons. Instead use the `equals` method with `case_sensitive=False`
```
Comment (amos-rimon) [inline pykusto/_src/expressions.py:499]: same here

File: pykusto/_src/expressions.py (hunk 6 above)
Comment (ymost) [reply pykusto/_src/expressions.py]: Fixed

File: pykusto/_src/expressions.py (hunk 7 above)
Comment (ymost) [reply pykusto/_src/expressions.py]: Fixed

**Review by amos-rimon — APPROVED** [review]

**Pull request merged**